code/AutoCompleteField.php

```diff
@@ -481,19 +488,16 @@ public function Suggest(SS_HTTPRequest $request)
             return;
         }
 
-        // Find field to search within
+        // Find fields to search within
         $sourceFields = $this->getSourceFields();
 
         // input
         $q = $request->getVar('term');
         $limit = $this->getLimit();
 
         $filters = array();
-
-        foreach (preg_split('/[\s,]+/', $q) as $keyword) {
-            foreach ($sourceFields as $sourceField) {
-                $filters["{$sourceField}:PartialMatch"] = $keyword;
-            }
+        foreach ($sourceFields as $sourceField) {
+            $filters["{$sourceField}:PartialMatch"] = $q;
```

Comment:
Currently if you search "new york" you will only get matches on "york". "new" is ignored because the `$filters["{$sourceField}:PartialMatch"]` item is over written in each pass of the `foreach (preg_split('/[\s,]+/', $q) as $keyword)` loop. It's filled at first with "new" and then overridden later with "york".

This change makes it match the full term "new york" only.